docs/src/ZIndexClasses.doc.js

```diff
@@ -108,4 +108,31 @@ function ZIndexBoxExample() {
   />
 );
 
+card(
+  <Stack gap={2}>
+    <Heading size="md">Note</Heading>
+    <Text>
+      FixedZIndex and CompositeZIndex work with Box and Sticky components. To
+      stay consistent across your codebase using zIndex classes, you can extract
+      zIndex values from both zIndex classes in cases where the zIndex receptor
+      does not accept zIndex classes.
```

Comment:
I'd call out here that this is an escape hatch that should only be used in cases like needing to work with a 3rd party library. The more "gestalt-y" way to handle this is wrap whatever needs a zIndex in a Box

Reply:
![image](https://user-images.githubusercontent.com/10593890/89692341-3a339180-d8c0-11ea-8922-63501ac449b6.png)